pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/ManualTransformFunctionMapProvider.java

```diff
@@ -0,0 +1,228 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.operator.transform.function;
+
+import java.util.EnumMap;
+import java.util.Map;
+import org.apache.pinot.common.function.TransformFunctionType;
+import org.apache.pinot.core.geospatial.transform.function.GeoToH3Function;
+import org.apache.pinot.core.geospatial.transform.function.GridDiskFunction;
+import org.apache.pinot.core.geospatial.transform.function.GridDistanceFunction;
+import org.apache.pinot.core.geospatial.transform.function.StAreaFunction;
+import org.apache.pinot.core.geospatial.transform.function.StAsBinaryFunction;
+import org.apache.pinot.core.geospatial.transform.function.StAsGeoJsonFunction;
+import org.apache.pinot.core.geospatial.transform.function.StAsTextFunction;
+import org.apache.pinot.core.geospatial.transform.function.StContainsFunction;
+import org.apache.pinot.core.geospatial.transform.function.StDistanceFunction;
+import org.apache.pinot.core.geospatial.transform.function.StEqualsFunction;
+import org.apache.pinot.core.geospatial.transform.function.StGeogFromGeoJsonFunction;
+import org.apache.pinot.core.geospatial.transform.function.StGeogFromTextFunction;
+import org.apache.pinot.core.geospatial.transform.function.StGeogFromWKBFunction;
+import org.apache.pinot.core.geospatial.transform.function.StGeomFromGeoJsonFunction;
+import org.apache.pinot.core.geospatial.transform.function.StGeomFromTextFunction;
+import org.apache.pinot.core.geospatial.transform.function.StGeomFromWKBFunction;
+import org.apache.pinot.core.geospatial.transform.function.StGeometryTypeFunction;
+import org.apache.pinot.core.geospatial.transform.function.StPointFunction;
+import org.apache.pinot.core.geospatial.transform.function.StPolygonFunction;
+import org.apache.pinot.core.geospatial.transform.function.StWithinFunction;
+
+
+/**
+ * This utility class provides a manual mapping of {@link TransformFunctionType} to the corresponding
+ * {@link TransformFunction} implementation classes.
+ *
+ * Ideally, this class should not exist. Instead, all transform functions should be registered using the
+ * {@link org.apache.pinot.core.udf.Udf} registration mechanism. But we need to keep this class for backward
+ * compatibility reasons until all transform functions are registered using the Udf mechanism.
+ */
+public class ManualTransformFunctionMapProvider {
```

Comment:
All this code was moved from TransformFunctionFactory to make that class easier to read